### PR TITLE
ref(sim_interface): AJD-697 improve zeromq port management - use a single-port

### PR DESCRIPTION
**Pull request by dmoszynski**
## Types of PR

- [ ] New Features
- [x] Upgrade of existing features
- [ ] Bugfix

## Link to the issue
https://tier4.atlassian.net/browse/AJD-697

## Description

- I've developed a `single-port` communication between `traffic_simulator` and `simple_sensor_simulator`. I've also investigated if this affects performance.

- In order to provide `single-port` communication:
  - I've add a [2 universal messages](https://github.com/RobotecAI/scenario_simulator_v2/blob/40e5fa6e8165e52fabd759489f821ef8ad61222a/simulation/simulation_interface/proto/simulation_api_schema.proto#L263-L302) based on Oneof proto message type [(at most one field will be set at the same time - it saves memory)](https://protobuf.dev/programming-guides/proto3/#oneof).  This provided a simple and memory-saving transfer of existing messages on a single port, their identification and easy deserialization:
  - I've adjusted [zmq_multi_client](https://github.com/RobotecAI/scenario_simulator_v2/blob/40e5fa6e8165e52fabd759489f821ef8ad61222a/simulation/simulation_interface/src/zmq_multi_client.cpp#L43-L199) to handle universal messages.
  - I've adjusted [zmq_mutli_server](https://github.com/RobotecAI/scenario_simulator_v2/blob/40e5fa6e8165e52fabd759489f821ef8ad61222a/simulation/simulation_interface/src/zmq_multi_server.cpp#L99-L180) to handle universal messages.
  - I've removed unnecessary ports (sockets) and their definitions.
  - I left a single port for communication, its value should be given as a rosparm `port` - for `simple_sensor_simulator_node` and `openscenario_interpreter_node`. ([commit](https://github.com/tier4/scenario_simulator_v2/commit/d4473010f5d3de910fac3340af2beecf97771637))

- I’ve developed a `test_zmq_node` ([reverted commit](https://github.com/tier4/scenario_simulator_v2/commit/83c309d2c9885c92221d298be70a692a2a318e3d)) which sends the same data as `traffic_simulator` and it allowed me to test performance in communication with `simple_sensor_simulator`.

### CODE REVIEW

File: .github/workflows/custom_spell.json [hunk 1]
```diff
@@ -23,4 +27,4 @@
     "TESTRANDOMIZER",
     "travelling"
   ]
-}
+}
```
Comment (hakuturu583) [inline .github/workflows/custom_spell.json:30]: Please add white line at EOF

File: simulation/simulation_interface/proto/simulation_api_schema.proto [hunk 2]
```diff
@@ -258,3 +258,45 @@ message UpdateTrafficLightsRequest {
...
+    SpawnPedestrianEntityResponse spawn_pedestrian_entity = 5;
+    SpawnMiscObjectEntityResponse  spawn_misc_object_entity = 6;
+    DespawnEntityResponse despawn_entity = 7;
+    UpdateEntityStatusResponse update_entity_status = 8;
+    AttachLidarSensorResponse attach_lidar_sensor = 9;
+    AttachDetectionSensorResponse attach_detection_sensor = 10;
+    AttachOccupancyGridSensorResponse attach_occupancy_grid_sensor = 11;
+    UpdateTrafficLightsResponse update_traffic_lights = 12;
+  }
+}
```
Comment (hakuturu583) [inline simulation/simulation_interface/proto/simulation_api_schema.proto:302]: Please add white line at EOF

**Review by hakuturu583 — CHANGES_REQUESTED** [review]
Code looks good to me, please update release note.

File: simulation/simulation_interface/proto/simulation_api_schema.proto (hunk 2 above)
Comment (dmoszynski) [reply simulation/simulation_interface/proto/simulation_api_schema.proto]: done

**Review by hakuturu583 — APPROVED** [review]

**Pull request merged**